pkg/defaulting/cluster.go

```diff
@@ -96,6 +96,10 @@ func DefaultClusterSpec(ctx context.Context, spec *kubermaticv1.ClusterSpec, tem
 
 	// Enforce audit logging
 	if datacenter.Spec.EnforceAuditLogging {
+		if spec.AuditLogging != nil {
+			spec.AuditLogging.Enabled = true
+		}
+
 		spec.AuditLogging = &kubermaticv1.AuditLoggingSettings{
 			Enabled: true,
 		}
```

Comment:
I don't think this fixes the override that you are trying to fix, given that the second part of the if branch still completely overrides `spec.AuditLogging`. I think what you want to do is this:

```suggestion
		if spec.AuditLogging == nil {
		    spec.AuditLogging = &kubermaticv1.AuditLoggingSettings{}
		}
		spec.AuditLogging.Enabled = true	
```

Reply:
good catch, thank you